### ISSUE TITLE
How to read json strings in array

### ISSUE THREAD

**Issue by polRk** [cdauth/json-stream-es#3]
I have an problem: i need to extract only json strings from response than contains array of simple objects:

```ts
new Response('[{}, {}, {}, ...]')...
// should return generator/stream of '{}', '{}', '{}'
```

**Comment by cdauth** [comment]
One way to do it would be to parse the JSON values and then stringify them again:
```javascript
import { parseJsonStream } from "json-stream-es";

const stream = response
	.pipeThrough(new TextDecoderStream())
	.pipeThrough(parseJsonStream([]))
	.pipeThrough(new TransformStream({
		transform: (chunk, controller) => {
			controller.enqueue(JSON.stringify(chunk));
		}
	}));
```

Another way (I’m not sure if it is more or less performant than the above approach) that would also preserve the formatting of the JSON would be to manually combine the different transformers to split up the stream into single JSON value streams and stringify those again:
```javascript
import { JsonParser, JsonPathDetector, JsonPathSelector, JsonPathStreamSplitter, streamToString } from "json-stream-es";

const stream = response
	.pipeThrough(new JsonParser())
	.pipeThrough(new JsonPathDetector())
	.pipeThrough(new JsonPathSelector((path) => path.length === 1))
	.pipeThrough(new JsonPathStreamSplitter())
	.pipeThrough(new TransformStream({
		transform: async (chunk, controller) => {
			controller.enqueue(await streamToString(chunk.pipeThrough(new JsonStringifier())));
		}
	});
```

I haven't tested this code, let me know if something doesn't work.

**Comment by polRk** [comment]
This solution is very unproductive, 1000 times slower than a regular stream

**Comment by cdauth** [comment]
Which of the two, and compared to what?

**Comment by cdauth** [comment]
Yes, if your data has a unique delimiter (which seems to be a newline in your case), it does not make any sense to use this library. This library parses the whole JSON, which is not necessary for your use case. It would be necessary only if there is no delimiter, so you don’t know where each JSON object ends. The example that you originally posted didn’t have a unique delimiter.

**Comment by polRk** [comment]
> so you don’t know where each JSON object ends

This problem is solved through the stack (For example, the reverse notation of writing an expression https://en.wikipedia.org/wiki/Polish_notation, https://en.wikipedia.org/wiki/Reverse_Polish_notation).

And what does JsonPathDetector do?

**Comment by cdauth** [comment]
> > so you don’t know where each JSON object ends
> 
> This problem is solved through the stack (For example, the reverse notation of writing an expression https://en.wikipedia.org/wiki/Polish_notation, https://en.wikipedia.org/wiki/Reverse_Polish_notation).

I don’t understand what you mean by that.


> And what does JsonPathDetector do?

It detects the path of each JSON chunk within the hierarchy of JSON values and is a prerequisite for `JsonPathSelector`. This is also documented in the README.
